### PR TITLE
:bug: Source Snapchat Marketing: remove deprecated auth specification

### PR DESCRIPTION
**Pull request by davydov-d**
## What
Remove deprecated `authSpecification` from connector spec in favour of `advancedAuth`
https://github.com/airbytehq/airbyte/issues/26246

## How
Replace `authSpecification` with `advancedAuth`

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]
Nice!

**Comment by davydov-d** [comment]
Removed Source Square from this PR to unblock it as Source Snapchat Marketing is blocked by invalid UI creds

File: airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/spec.json [hunk 1]
```diff
@@ -48,12 +48,42 @@
       }
     }
   },
-  "authSpecification": {
-    "auth_type": "oauth2.0",
-    "oauth2Specification": {
-      "rootObject": [],
-      "oauthFlowInitParameters": [["client_id"], ["client_secret"]],
-      "oauthFlowOutputParameters": [["refresh_token"]]
+  "advanced_auth": {
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/spec.json:51]: as soon as we have access to our account - I'll be able to confirm everything works as expected via manual testing on localhost

**Review by maxi297 — APPROVED** [review]
Conditional approval given that we have ensured that refresh_token, client_id and client_secret are passed to the new auth spec (see [this](https://github.com/airbytehq/airbyte/pull/26358#discussion_r1204252635))

**Pull request merged**